File: src/main/java/pl/coddlers/core/models/entity/CourseEditionLesson.java
package pl.coddlers.core.models.entity;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Data
@NoArgsConstructor
@ToString(exclude={"courseEdition", "lesson"})
public class CourseEditionLesson {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private Timestamp startDate;
    private Timestamp endDate;

    @ManyToOne(fetch = FetchType.LAZY, targetEntity = CourseEdition.class)
    @JoinColumn(name = "course_edition_id")
    private CourseEdition courseEdition;

    @ManyToOne(fetch = FetchType.LAZY, targetEntity = Lesson.class)
    @JoinColumn(name = "lesson_id")
    private Lesson lesson;
}

File: src/main/webapp/src/app/models/courseEditionLesson.ts
export class CourseEditionLesson {
  private _id: number;
  private _startDate: Date;
  private _endDate: Date;
  private _courseEditionId: number;
  private _lessonId: number;

  constructor(id: number, startDate: Date, endDate: Date, courseEditionId: number, lessonId: number) {
    this._id = id;
    this._startDate = startDate;
    this._endDate = endDate;
    this._courseEditionId = courseEditionId;
    this._lessonId = lessonId;
  }

  get id(): number {
    return this._id;
  }

  get startDate(): Date {
    return this._startDate;
  }

  get endDate(): Date {
    return this._endDate;
  }

  get courseEditionId(): number {
    return this._courseEditionId;
  }

  get lessonId(): number {
    return this._lessonId;
  }

  public static fromJSON(jsonObj: any): CourseEditionLesson {
    return new CourseEditionLesson(+jsonObj.id, new Date(jsonObj.startDate), new Date(jsonObj.endDate),
      +jsonObj.courseEditionId, +jsonObj.lessonId);
  }

  public toJSON(): any {
    return {
      id: this.id,
      startDate: this.startDate,
      endDate: this.endDate,
      courseEditionId: this.courseEditionId,
      lessonId: this.lessonId
    };
  }
}
